### ISSUE TITLE
"Unexpected token {" when running 'gulp' default command for the first time

### ISSUE THREAD

**Issue by richiethomas** [FreeCodeCamp/FreeCodeCamp#11044]
#### Issue Description

I just finished downloading and setting up my local copy of the freeCodeCamp codebase.  I followed the instructions in the CONTRIBUTING.md file, including running a global install of both gulp and bower, and running "npm install" and "bower install".  However, when I'm asked to run the "gulp" default command, I get the following error:
```

/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/gulpfile.js:219
      reqHeaders: ({ url: { hostname } }) => ({
                   ^

SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:387:25)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Liftoff.handleArguments (/usr/local/lib/node_modules/gulp/bin/gulp.js:116:3)
    at Liftoff.<anonymous> (/usr/local/lib/node_modules/gulp/node_modules/liftoff/index.js:198:16)
    at module.exports (/usr/local/lib/node_modules/gulp/node_modules/flagged-respawn/index.js:17:3)
```

I noticed in the error message that it's somehow related to line 219 of gulpfile.js, which just happens to be the first time that ES6-specific syntax is used in that file (everything before that is just vanilla JS).  I had a theory that it was something to do with ES6 compilation, so I renamed the gulpfile to "gulpfile-temp.js" and ran babel on this new file, which generated a vanilla JS version of the original gulpfile.  I was then able to run the "gulp" default command with some success, however attempting to navigate to localhost:3001 results in a crash of my Chrome browser for some reason.  I noticed other endpoints mentioned in the server logs (i.e. "localhost:3001/explorer", "localhost:3002", and "localhost:1337"), all 3 of which worked fine.

I'm curious about the following:

-Should instructions be added to the CONTRIBUTING.md file which help new developers compile the gulpfile.js into vanilla JS, so it can be run?  Or is there something unique to my machine that caused this problem?
-Why does localhost:3001 crash my browser?  Given that this endpoint is mentioned as the first URL you should access after you start up the server, has anyone else encountered this problem or is it just me?

#### Browser Information

* Browser Name, Version: Chrome 53.0.2785.143
* Operating System: OSX v10.11.5
* Mobile, Desktop, or Tablet: Desktop
Gulp: [21:51:17] CLI version 3.9.1, [21:51:17] Local version 3.9.1
Bower: 1.7.9

UPDATE: localhost:3001/explorer no longer seems to be working, for some reason.

UPDATE #2: When I run the 'gulp' command, and before I try to access localhost:3001, I see the following "Address In Use" error included among the output in the gulp server logs:

```
[BS] Proxying: http://localhost:3001
[BS] Access URLs:
 ------------------------------------
       Local: http://localhost:3001
    External: http://172.20.10.4:3001
 ------------------------------------
          UI: http://localhost:3002
 UI External: http://172.20.10.4:3002
 ------------------------------------
[BS] Watching files...
  fcc:user:remote setting up user hooks +0ms
Error: listen EADDRINUSE :::3001
    at Object.exports._errnoException (util.js:890:11)
    at exports._exceptionWithHostPort (util.js:913:20)
    at Server._listen2 (net.js:1234:14)
    at listen (net.js:1270:10)
    at Server.listen (net.js:1366:5)
    at EventEmitter.app.listen (/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/node_modules/loopback/lib/application.js:585:19)
    at EventEmitter.<anonymous> (/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/server/server.js:50:7)
    at EventEmitter.start (/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/node_modules/lodash/lodash.js:9981:25)
    at Object.<anonymous> (/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/server/server.js:70:7)
    at Module._compile (module.js:413:34)
    at loader (/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at /Users/richiethomas/Desktop/Workspace/OpenSource/freecodecamp/node_modules/babel-cli/lib/_babel-node.js:159:24
[21:47:38] [nodemon] app crashed - waiting for file changes before starting...
webpack built 45f0f27d44eca7a8bd6d in 14403ms

WARNING in ./~/babel-core/lib/transformation/file/options/option-manager.js
Critical dependencies:
174:19-37 the request of a dependency is an expression
301:16-34 the request of a dependency is an expression
 @ ./~/babel-core/lib/transformation/file/options/option-manager.js 174:19-37 301:16-34

WARNING in ./~/babel-core/lib/transformation/file/index.js
Critical dependencies:
515:24-39 the request of a dependency is an expression
714:16-34 the request of a dependency is an expression
 @ ./~/babel-core/lib/transformation/file/index.js 515:24-39 714:16-34
webpack: bundle is now VALID.

```

But when I run "lsof -i :3001" in another terminal window, the only process using port 3001 is the freeCodeCamp 'gulp' command:
```

MacBook-Pro-4:freecodecamp richiethomas$ lsof -i :3001
COMMAND   PID         USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
node    96583 richiethomas   22u  IPv6 0xbaa656e55f949c6f      0t0  TCP *:redwood-broker (LISTEN)
```

#### Your Code

```js
 
 
 
```

**Comment by dhcodes** [comment]
@richiethomas this may be caused by your versions of npm and node. Make sure they are 3+ and 6+ respectively. Then, after you've updated both, in your local directory of freecodecamp, run `rm -rf node_modules` and then `npm install`. Typically localhost:3000 will be the address when gulp is up and running correctly. Checkout the [Contributors](https://gitter.im/FreeCodeCamp/Contributors) room on Gitter for more assistance.

**Comment by richiethomas** [comment]
Yep, turns out it was a Node version issue.  I had followed the steps to upgrade to Node 6, but I hadn't included the following code in my .bash_profile (as indicated in the output from my NVM installation):

export NVM_DIR="/Users/richiethomas/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"  # This loads nvm

So every time I opened a new command line tab, my Node version was reset to 5.9.1.  And I'm guessing there's an important difference in the way ES6 function declarations are handled in Node v6 vs. v5.9.1.
